src/main/java/net/io/kino/service/impl/AuthenticationManagerImpl.java

```diff
@@ -0,0 +1,80 @@
+package net.io.kino.service.impl;
+
+import net.io.kino.repository.ManagersRepository;
+import net.io.kino.model.User;
+import net.io.kino.service.AuthenticationManager;
+import org.apache.tomcat.util.codec.binary.Base64;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import java.security.SecureRandom;
+
+public class AuthenticationManagerImpl implements AuthenticationManager {
+
+    private static final int iterations = 20*1000;
+    private static final int saltLen = 32;
+    private static final int desiredKeyLen = 256;
+
+    @Autowired
+    private ManagersRepository managers;
+
+    @Override
+    public boolean login(User user) throws Exception {
+        if(authenticate(user)) {
+            user.setStatus(true);
+            return true;
+        }
+        else {
+            user.setStatus(false);
+            return false;
+        }
+
+    }
+
+    @Override
+    public boolean logout(User user) {
+        if(user.getStatus()){
+            user.setStatus(false);
+            return true;
+        }
+        else{
+            return false;
+        }
+    }
+
+    // check if hash of a given password corresponds with a password in database
+    @Override
+    public boolean authenticate(User user) throws Exception {
+        User found = managers.getOne(user.getID());
+        String stored = found.getPassword();
+
+        String[] saltAndHash = stored.split("\\$");
+        if (saltAndHash.length != 2) {

```

Comment:
Jeśli pobieramy dane z bazy to chyba zakładamy, że są już poprawne.

Reply:
Wiesz, przezorny zawsze ubezpieczony. Mogłaby się pojawić sytuacja, że ktoś tam coś ręcznie wpisał. Ja zawsze wolałam wszystko dwa razy zabezpieczać. Jeśli uważacie, że to za dużo, to można to wywalić, ale jak dla mnie to zawsze lepiej się upewnić, że pobrane dane są poprawne